packages/front-end/services/sqlFormatter.ts

```diff
@@ -0,0 +1,83 @@
+import { DataSourceType } from "back-end/types/datasource";
+import { format } from "shared/sql";
+import { FormatDialect } from "shared/src/types";
+
+export function canFormatSql(datasourceType: DataSourceType): boolean {
+  return !!getSqlDialect(datasourceType);
+}
+
+function getSqlDialect(datasourceType: DataSourceType): FormatDialect | "" {
```

Comment:
Not all data source types have a dialect. So in those cases, we return an empty string so we don't try to format the query with the wrong language when we call `sql-formatter`'s `format` function.